builder/ecs/step_config_eip.go

```diff
@@ -31,59 +31,63 @@ func (s *stepConfigAlicloudEIP) Run(ctx context.Context, state multistep.StateBa
 	ui := state.Get("ui").(packersdk.Ui)
 	instance := state.Get("instance").(*ecs.Instance)
 
-	if s.SSHPrivateIp {
-		ipaddress := instance.VpcAttributes.PrivateIpAddress.IpAddress
-		if len(ipaddress) == 0 {
-			ui.Say("Failed to get private ip of instance")
-			return multistep.ActionHalt
-		}
-		state.Put("ipaddress", ipaddress[0])
-		return multistep.ActionContinue
-	}
+	var ipaddress string
 
-	ui.Say("Allocating eip...")
+	if s.AssociatePublicIpAddress || s.InternetChargeType != "" || s.InternetMaxBandwidthOut >= 0 {
```

Comment:
Is it safe to assume that if InternetChargeType or InternetMaxBadnwidthOut is set that an EIP should b allocated? 

It seems counter to the AssociatePublicIpAddess configuration. Unless setting one of the two options sets AssociatePublicIpAddress to true, which quickly looking at the code doesn't seem to be the case. 

If this would be a surprise to the user I would suggest the following
```suggestion
	if s.AssociatePublicIpAddress {
```

Reply:
For me, it's fine to think that setting this two options would assume there will be a EIP allocated.
But what you mentioned is also a good concern.